app/src/androidTest/java/org/schabi/newpipe/local/history/HistoryRecordManagerTest.kt

```diff
@@ -97,14 +97,22 @@ class HistoryRecordManagerTest {
         assertThat(database.searchHistoryDAO().all.blockingFirst()).isEmpty()
     }
 
-    @Test
-    fun getRelatedSearches_emptyQuery() {
+    private fun insertShuffledRelatedSearches(relatedSearches: Collection<SearchHistoryEntry>) {
+
+        // shuffle to make sure the order of items returned by queries depends only on
+        // SearchHistoryEntry.creationDate, not on the actual insertion time
+        database.searchHistoryDAO().insertAll(relatedSearches.shuffled())
```

Comment:
This is a very subjective view, so feel free to ignore it:

I finally figured out, why this stood out to me. It ensures that the `ORDER_BY_CREATION_DATE` inside the dao is working but does it implicitly in every test instead of getting its own dedicated test. The 2 lines of comments are needed to convey this implicit test. In my personal opinion this should be put into its own test and then the helper method can also be removed as I see 2 lines as an acceptaple duplication

Reply:
Nice catch. I think order should be enforced in each test, since each test checks that the correct items are returned and in the correct order in many situations. Therefore I'll leave the code as it is and add a comment explaining that shuffling the items is used to verify the `ORDER BY` works. I pushed, please approve again ;-)